backend/src/main/java/io/github/projektalmanac/amoxtli/backend/service/UserService.java

```diff
@@ -342,21 +354,15 @@ public IntercambioDto solicitaIntercambio(Integer idOfertante, CreacionIntercamb
         Integer idAceptante = creacionIntercambioDto.getIdAceptante();
         Integer idLibroAceptante = creacionIntercambioDto.getIdLibroAceptante();
 
-        if (Objects.equals(idOfertante, idAceptante)){
-            throw new BadRequestException("No puedes intercambiar libros contigo mismo.");
+        if (idOfertante.equals(idAceptante)) {
+            throw new SelfExchangeException(idOfertante);
         }
         // Obtenemos a los usuarios y al libro que se intercambiará
-        Optional <User> userOptionalOfertante = this.userRepository.findById(idOfertante);
-        if (userOptionalOfertante.isEmpty()) {
-            throw new UserNotFoundException(idOfertante);
-        }
 
-        Optional<User> userOptionalAceptante = this.userRepository.findById(idAceptante);
-        if (userOptionalAceptante.isEmpty()) {
-            throw new UserNotFoundException(idAceptante);
-        }
+        Optional<User> userOptionalOfertante = Optional.of(this.userRepository.findById(idOfertante)).orElseThrow(() -> new UserNotFoundException(idOfertante));
+        Optional<User> userOptionalAceptante = Optional.of(this.userRepository.findById(idAceptante)).orElseThrow(() -> new UserNotFoundException(idAceptante));
```

Comment:
```suggestion
        User userOfertante = this.userRepository.findById(idOfertante)).orElseThrow(() -> new UserNotFoundException(idOfertante);
        User userAceptante = this.userRepository.findById(idAceptante)).orElseThrow(() -> new UserNotFoundException(idAceptante);
```

El punto de hacer `orElseThrow` es obtener al objeto `User` y dejar de usar `Optional`